SparseVector/Task/SparseVector.cs

```diff
@@ -0,0 +1,223 @@
+    /// </summary>
+    /// <param name="size">Size of the vector.</param>
+    public SparseVector(int size)
+    {
+        this.size = size;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="vector">List, that contains pairs of (index, number) that is going to be contained in sparse vector.</param>
+    /// <param name="size">Size of the sparse vector.</param>
+    public SparseVector(List<(int, int)> vector, int size)
+    {
+        this.size = size;
+        foreach (var element in vector)
+        {
+            if (element.Item1 >= size)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            if (element.Item2 == 0)
+            {
+                continue;
+            }
+
+            if (this.vector.ContainsKey(element.Item1))
+            {
+                throw new ArgumentException("Initializer vector contained same index more than once.");
+            }
+
+            this.vector.Add(element.Item1, element.Item2);
+        }
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="givenVector">Vector with numbers that is going to be added to sparse vector.</param>
+    /// <param name="size">Size of the vector.</param>
+    /// <exception cref="IndexOutOfRangeException">Being thrown if given vector`s size is bigger than size of sparse vector. </exception>
+    public SparseVector(List<int> givenVector, int size)
+    {
+        if (givenVector.Count > size)
+        {
+            throw new IndexOutOfRangeException();
+        }
+
+        this.size = size;
+        for (int i = 0; i < givenVector.Count; i++)
+        {
+            if (givenVector[i] == 0)
+            {
+                continue;
+            }
+
+            this.vector.Add(i, givenVector[i]);
+        }
+    }
+
+    /// <summary>
+    /// Gets size of the vector.
+    /// </summary>
+    public int Size { get => this.size; }
+
+    /// <summary>
+    /// Gets a value indicating whether vector is empty.
+    /// </summary>
+    public bool IsNull { get => this.vector.Count == 0; }
+
+    /// <summary>
+    /// Gets or sets value by index.
+    /// </summary>
+    /// <param name="index">Index of the element in sparse vector.</param>
+    /// <returns>Value contained by the sparse vector by given index.</returns>
+    /// <exception cref="IndexOutOfRangeException">Throws if given index is greater than size of the sparse vector.</exception>
+    public int this[int index]
+    {
+        get
+        {
+            if (index >= this.size)
+            {
+                throw new IndexOutOfRangeException(nameof(index));
+            }
+
+            if (this.vector.ContainsKey(index))
+            {
+                return this.vector[index];
+            }
+            else
+            {
+                return 0;
+            }
```

Comment:
```suggestion
            return this.vector.ContainsKey(index) ? this.vector[index] : 0;
```
А то как на Паскале программируете.